packages/interbit-ui-components/src/tests/reduxForm.test.js

```diff
@@ -0,0 +1,72 @@
+import assert from 'assert'
+import validation from '../components/ReduxForm/validation'
+
+describe('ReduxForm component', () => {
+  describe('IbField component', () => {
+    // TODO #694: implement the following tests
+    it('renders an <input type="text" /> when type="text"', () => {})
+    it('renders a <select /> and <option />s when type="select"', () => {})
+    it('renders a <select multiple /> and <option />s when type="select" and multiple={true}', () => {})
+    it('renders an <input type="checkbox" /> when type="checkbox"', () => {})
+    it('renders an <input type="radio" /> when type="radio"', () => {})
+  })
+
+  describe('Validation functions', () => {
+    it('checks for valid email formats in validation.email', () => {
+      const validEmails = ['meow@meow.com', '', undefined, null]
+      const invalidEmails = [
+        'meow',
+        '@meow',
+        '.com',
+        '@meow.com',
+        'meow@meow',
+        false,
+        true,
+        0,
+        {}
+      ]
+
+      validEmails.forEach(email => {
+        assert.strictEqual(validation.email(email), undefined)
+      })
+      invalidEmails.forEach(email => {
+        assert.strictEqual(validation.email(email), 'Invalid email address.')
+      })
+    })
+
+    it('checks for finite numbers in validation.number', () => {
+      const validValues = [0, 100, 100.0]
+      const invalidValues = [
+        false,
+        true,
+        'meow',
+        undefined,
+        null,
+        [],
+        {},
+        Number.POSITIVE_INFINITY,
+        Number.NEGATIVE_INFINITY,
+        '100'
```

Comment:
Add empty string and NaN to your tests. For other test cases see `privateCovenant.test.js` in `interbit-template`

See comments about testing numbers entered into a form. I think what we care about, is that the value entered into the form can be interpreted as a number, which means that '100' is valid.

Reply:
👍